x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/DenseVectorFieldTypeIT.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.action.index.IndexRequestBuilder;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xpack.esql.action.AbstractEsqlIntegTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.index.IndexSettings.INDEX_MAPPER_SOURCE_MODE_SETTING;
+import static org.elasticsearch.index.mapper.SourceFieldMapper.Mode.SYNTHETIC;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+public class DenseVectorFieldTypeIT extends AbstractEsqlIntegTestCase {
+
+    private static final Set<String> DENSE_VECTOR_INDEX_TYPES = Set.of(
+        "int8_hnsw",
+        "hnsw",
+        "int4_hnsw",
+        "bbq_hnsw",
+        "int8_flat",
+        "int4_flat",
+        "bbq_flat",
+        "flat"
+    );
+
+    private final String indexType;
+    private final boolean index;
+    private final boolean synthetic;
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() throws Exception {
+        List<Object[]> params = new ArrayList<>();
+        // Indexed field types
+        for (String indexType : DENSE_VECTOR_INDEX_TYPES) {
+            params.add(new Object[] { indexType, true, false });
+        }
+        // No indexing
+        params.add(new Object[] { null, false, false });
+        // No indexing, synthetic source
+        params.add(new Object[] { null, false, true });
+        return params;
+    }
+
+    public DenseVectorFieldTypeIT(@Name("indexType") String indexType, @Name("index") boolean index, @Name("synthetic") boolean synthetic) {
+        this.indexType = indexType;
+        this.index = index;
+        this.synthetic = synthetic;
+    }
+
+    private final Map<Integer, List<Float>> indexedVectors = new HashMap<>();
+
+    public void testRetrieveFieldType() {
+        var query = """
+            FROM test
+            """;
+
+        try (var resp = run(query)) {
+            assertColumnNames(resp.columns(), List.of("id", "vector"));
+            assertColumnTypes(resp.columns(), List.of("integer", "dense_vector"));
```

Comment:
are we missing some assertions here for the values? to at least check that they are not nulls?

Reply:
That's done in other tests - I started with this test to just check that the field types are retrieved correctly.